### PR TITLE
Wallet public address

### PR DESCRIPTION
**Pull request by AlexanderSuprunenko**

### CODE REVIEW

File: src/supernode/supernode.cpp [hunk 1]
```diff
@@ -50,10 +54,76 @@ bool Supernode::initConfigOption(int argc, const char** argv, ConfigOpts& config
...
+    if (!boost::filesystem::exists(wallet_keys_file))
+    {
+        LOG_PRINT_L0("file '") << wallet_keys_file << "' not found. Generating the keys";
+        crypto::public_key W;
+        crypto::generate_keys(W, w);
+        //save secret key
+        boost::filesystem::path wallet_keys_file_tmp = wallet_keys_file;
+        wallet_keys_file_tmp += ".tmp";
+        w_str = epee::string_tools::pod_to_hex(w);
+        bool r = epee::file_io_utils::save_string_to_file(wallet_keys_file_tmp.string(), w_str);
```
Comment (bitkis) [inline src/supernode/supernode.cpp]: what happens if the tmp file exists?

File: include/supernode/supernode.h [hunk 2]
```diff
@@ -16,11 +16,13 @@ class Supernode : public GraftServer
         // runtime parameters.
         // path to watch-only wallets (supernodes)
         std::string watchonly_wallets_path;
+        std::string wallet_public_address;
         // testnet flag
         bool testnet;
     };
 
     void prepareDataDirAndSupernodes();
+    void prepareWalletKey(std::string& id_key);
```
Comment (mbg033) [inline include/supernode/supernode.h]: why don't just return it?
`std::string prepareWalletKey()`

**Comment by jagerman** [comment]
Why is this PR needed at all?  It seems completely redundant: the staking address is already tied to the supernode via the staking transaction in the blockchain (and the secondary database built from that).  Having it in the config file here doesn't seem useful.

**Comment by AlexanderSuprunenko** [comment]
Another one created instead
https://github.com/graft-project/graft-ng/pull/202

**Pull request closed without merging**